docs/premium-features/custom-types-and-actions.md

```diff
@@ -0,0 +1,259 @@
+:::
+
+### Create Action
+
+To create a new Action, follow these steps:
+
+#### Module Name and Description
+
+1. Click on the "Add Action" button in the top right of the APIs tab.
+2. Provide a Display Name for your new Action, such as "My New Module Name". This will automatically populate the Name field with a generated name like "MyNewModuleName".
+3. Optionally, add a description for your Action.
+
+:::note
+If you create a new Action from the "All Modules" screen, you can choose which Module the Action will reside in.
+:::
+
+#### Choosing Types
+
+1. In the "Types" section, select the input and output types for your Action:
+   - For the Input Type, choose the type of the property from the dropdown (e.g., "String", "Number", "Boolean", "Date", "JSON", "DTO", "Enum").
+   - If you choose "DTO", click "Select..." to pick an existing DTO or create a new one.
+   - Repeat the same steps for the Output Type.
+2. If needed, you can change the Input Type or Output Type to an array by toggling the "Array" switch.
+
+#### REST API Settings
+
+1. Choose the appropriate HTTP Method from the dropdown (e.g., "get", "post", "put", "delete").
+2. Specify the Path for your Action (e.g., "/id/my-new-module-name").
+3. Select the Input Source for your Action's parameters (e.g., "body", "query", "params", or "split") from the dropdown.
+
+#### GraphQL API Settings
+
+Choose the Operation Type for your Action (e.g., "query" or "mutation").
+
+![Create Action](./assets/custom-types-and-actions/create-action.png)
+
+### Enable and Disable Actions
+
+Amplication gives you a complete overview of all Actions associated with a specific Module.
+
+This is useful when you want to temporarily disable certain endpoints without completely removing them from your application.
+
+To enable or disable an Action, simply toggle the switch next to the Action's name in the APIs tab.
+
+:::caution
+Disabling an Action will cause Amplication not to generate the code for that Action.
+:::
+
+![Enable and Disable Actions](./assets/custom-types-and-actions/all-actions.png)
+
+### Edit and Delete Action
+
+To edit an Action, click on the Action within its associated Module in the APIs tab. You can modify the Action's properties, such as the HTTP method, path, input DTO, output DTO, and other settings as needed.
+
+To delete an Action, click on the delete button next to the Action's name. Be cautious when deleting Actions, as it may impact other parts of your application that depend on them.
+
+:::note
+Actions generated by Amplication **cannot be deleted**.
+:::
+
+## Conclusion
+
+Custom Types & Actions in Amplication provide a powerful and flexible way to define and manage your application's data structures and API endpoints. By leveraging custom DTOs and Actions, you can create a tailored and efficient API that meets your specific requirements and generates the production-ready code for it in your repository.
```

Comment:
@PazYanoverr, perhaps elaborate here on benefits like a single source of truth for APIs and DTOs, fast creation of all your APIs, and standardization.

Reply:
Agree, great points!
@dericksozo - Let's emphasize it not only at the conclusion but also at the beginning of the doc.